CoreScriptsRoot/Modules/Settings/Utility.lua

```diff
@@ -2017,4 +2127,7 @@ function moduleApiTable:TweenProperty(instance, prop, start, final, duration, ea
 	return PropertyTweener(instance, prop, start, final, duration, easingFunc, cbFunc)
 end
 
+---------------------- Main code -----------------------
+RunService.RenderStepped:connect(DetectCustomButtonHoverStates)
```

Comment:
This is a CRITICAL problem:
Utility is for various helper and math functions to cut-down on commonly re-used code. We cannot be connecting to global scope events here, especially ones like RenderStepped which happens at a high frequency.